pkg/cmd/cli/backuplocation/delete.go

```diff
@@ -0,0 +1,121 @@
+                velero backup-location delete backup-location-1
+        
+                # Delete a backup storage location named "backup-location-1" without prompting for confirmation.
+                velero backup-location delete backup-location-1 --confirm
+        
+                # Delete backup storage locations named "backup-location-1" and "backup-location-2."
+                velero backup-location delete backup-location-1 backup-location-2
+        
+                # Delete all backup storage locations labelled with foo=bar."
+                velero backup-location delete --selector foo=bar
+                
+                # Delete all backup storage locations.
+                velero backup-location delete --all`,
+
+                Run: func(c *cobra.Command, args []string) {
+                        cmd.CheckError(o.Complete(f, args))
+                        cmd.CheckError(o.Validate(c, f, args))
+                        cmd.CheckError(Run(o))
+                },
+        }
+
+        o.BindFlags(c.Flags())
+        return c
+}
+
+// Run performs the delete backup-location operation.
+func Run(o *cli.DeleteOptions) error {
+        if !o.Confirm && !cli.GetConfirmation() {
+                // Don't do anything unless we get confirmation
+                return nil
+        }
+
+        var (
+                backupLocations []*velerov1api.BackupStorageLocation
+                errs    []error
+        )
+
+        switch {
+        case len(o.Names) > 0:
+                for _, name := range o.Names {
+                        backupLocation, err := o.Client.VeleroV1().BackupStorageLocations(o.Namespace).Get(context.TODO(), name, metav1.GetOptions{})
+                        if err != nil {
+                                errs = append(errs, errors.WithStack(err))
+                                continue
+                        }
+                        backupLocations = append(backupLocations, backupLocation)
+                }
+        default:
+                selector := labels.Everything().String()
+                if o.Selector.LabelSelector != nil {
+                        selector = o.Selector.String()
+                }
+                res, err := o.Client.VeleroV1().BackupStorageLocations(o.Namespace).List(context.TODO(), metav1.ListOptions{
+                        LabelSelector: selector,
+                })
+                if err != nil {
+                        errs = append(errs, errors.WithStack(err))
+                }
+
+                for i := range res.Items {
+                        backupLocations = append(backupLocations, &res.Items[i])
+                }
+        }
+        if len(backupLocations) == 0 {
+                fmt.Println("No backup-locations found")
+                return nil
+        }
+
+        for _, s := range backupLocations {
+                err := o.Client.VeleroV1().BackupStorageLocations(s.Namespace).Delete(context.TODO(), s.Name, metav1.DeleteOptions{})
```

Comment:
We should also delete all the `backup` objects from the backup storage location before deleting the backup storage location.